packages/core/src/modules/sd-jwt-vc/credential-status/token-status-list/TokenStatusList.ts

```diff
@@ -0,0 +1,237 @@
+  uri: string
+  indices: number[]
+  publish: boolean
+}
+
+/**
+ * @internal
+ */
+@injectable()
+export class TokenStatusListService {
+  private sdJwtVcModuleConfig: SdJwtVcModuleConfig
+
+  public constructor(sdJwtVcModuleConfig: SdJwtVcModuleConfig) {
+    this.sdJwtVcModuleConfig = sdJwtVcModuleConfig
+  }
+
+  async createStatusList(
+    agentContext: AgentContext,
+    issuer: SdJwtVcIssuer,
+    options: CreateTokenStatusListOptions
+  ): Promise<CreateTokenStatusListResult> {
+    const sdJwtService = agentContext.dependencyManager.resolve(SdJwtVcService)
+    const jwsService = agentContext.dependencyManager.resolve(JwsService)
+
+    const statusList = new StatusList(new Array(options.size).fill(0), 1)
+
+    const issuerKey = await sdJwtService.extractKeyFromIssuer(agentContext, issuer, true)
+
+    // construct jwt payload
+    const jwtPayload = new JwtPayload({
+      iss: issuerKey.iss,
+      iat: Math.floor(Date.now() / 1000),
+      additionalClaims: {
+        status_list: {
+          bits: statusList.getBitsPerStatus(),
+          lst: statusList.compressStatusList(),
+        },
+      },
+    })
+
+    // sign JWT
+    const jwt = await jwsService.createJwsCompact(agentContext, {
+      payload: jwtPayload,
+      keyId: issuerKey.publicJwk.keyId,
+      protectedHeaderOptions: {
+        alg: issuerKey.alg,
+        typ: 'statuslist+jwt',
+        kid: issuerKey.kid,
+      },
+    })
+
+    if (options.publish) {
+      const result = await this.publishStatusList(agentContext, issuer, jwt, options)
+      return { jwt, uri: result }
+    }
+
+    return { jwt }
+  }
+
+  async revokeIndex(
+    agentContext: AgentContext,
+    issuer: SdJwtVcIssuer,
+    options: RevokeIndicesOptions
+  ): Promise<boolean> {
+    const sdJwtService = agentContext.dependencyManager.resolve(SdJwtVcService)
+    const jwsService = agentContext.dependencyManager.resolve(JwsService)
+
+    const { indices, publish, ...publishOptions } = options
+
+    const currentStatusListJwt = await this.getStatusList(agentContext, options.uri)
+    const parsedStatusListJwt = Jwt.fromSerializedJwt(currentStatusListJwt)
+
+    // extract and validate iss
+    const iss = parsedStatusListJwt.payload.iss as string
+    if (issuer.method === 'did' && parseDid(issuer.didUrl).did !== iss) {
+      throw new TokenStatusListError('Invalid issuer')
+    }
+    if (issuer.method === 'x5c' && issuer.issuer !== iss) {
+      throw new TokenStatusListError('Invalid issuer')
+    }
+
+    const header = parsedStatusListJwt.header
+
+    const statusList = getListFromStatusListJWT(currentStatusListJwt)
+    // update indices
+    for (const revokedIndex of options.indices) {
+      statusList.setStatus(revokedIndex, 1)
+    }
+
+    const issuerKey = await sdJwtService.extractKeyFromIssuer(agentContext, issuer, true)
+
+    // construct jwt payload
+    const jwtPayload = new JwtPayload({
+      iss,
+      iat: Math.floor(Date.now() / 1000),
```

Comment:
Maybe we can move `dateToSeconds` from openid4vc package to core utils and use that?